### PR TITLE
ValidateUsername proof rpc

### PR DESCRIPTION
**Pull request by joshblum**
RPC to validate username for param/nonparam proofs

### CODE REVIEW

File: go/externals/proof_service_facebook.go [hunk 1]
```diff
@@ -55,13 +55,7 @@ func (t *FacebookServiceType) NormalizeUsername(s string) (string, error) {
...
 	// Allow a leading '@'.
 	s = strings.TrimPrefix(s, "@")
-	if !facebookUsernameRegexp.MatchString(s) {
-		return "", libkb.NewBadUsernameError(s)
-	}
-	// This is the normalization function that gets called by the Prove engine.
-	// Avoid stripping dots, so that we can preserve them when the username is
-	// displayed.
-	return strings.ToLower(s), nil
+	return t.NormalizeUsername(s)
```
Comment (mlsteele) [inline go/externals/proof_service_facebook.go]: something change about the dots?

Comment (joshblum) [reply go/externals/proof_service_facebook.go]: nope!

Comment (mlsteele) [reply go/externals/proof_service_facebook.go]: oh i see. great

Comment (joshblum) [reply go/externals/proof_service_facebook.go]: 🎉

**Pull request merged**